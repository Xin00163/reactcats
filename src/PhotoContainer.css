
body {
  margin: 0;
  padding: 0;
  font-family: helvetica;
}

.logo {
  font-size: 60px;
}

.PhotoContainer {
  list-style-type: none;
  margin: 1rem 0;
  padding: 1rem 0;
  border: 1px solid #eee;
}

.User {
  display: flex;
  margin-bottom: 1rem;
  padding: 0 1rem;
}

.Username {
  margin: 0;
  padding: 20px;
  text-align: left;
}

.Avatar {
  object-fit: cover;
  width: 64px;
  height: 64px;
  border-radius: 100%;
}

.Image {
  margin: 0 auto;
  width: 100%
}

.Like {
  display: flex;
  justify-content: flex-start;
  padding: 1rem;
}

#Click-like {
  margin-right: 0.5rem;
}

.fa-heart {
 color: #f52f2f;
}

.fa-heart-o {
  color: #666;
}

.Description {
  padding: 0.5rem 1rem;
  text-align: left;
}
