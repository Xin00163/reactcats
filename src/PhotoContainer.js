import React, { Component } from 'react';
import images from './images';
import avatars from './avatars';
import './PhotoContainer.css';

class PhotoContainer extends Component {

  constructor(props) {
    super(props);
    this.imageLink = this.props.data.imageLink.split('.')[0];
    this.userIconLink = this.props.data.userIconLink.split('.')[0];

    this.state = {
      likes: this.props.data.likes,
      liked: this.props.data.liked,
      showPost: true,
    };
    this.likeClick = this.likeClick.bind(this);
  }


  likeClick(e) {
    e.preventDefault();
    this.setState(prevState => ({
      likes: prevState.liked ? prevState.likes - 1 : prevState.likes + 1,
      liked: !prevState.liked
    }));
  }


  render() {
    return (
      <div className={'PhotoContainer' + (this.state.showPost ? '' : ' deleted')}>
        <button id='Click-delete' onClick={this.onDeleteClick}>Delete post</button>
        <div className='User'>
          <img className='Avatar' src={avatars[this.userIconLink]} />
          <h2 className='Username'>{this.props.data.username}</h2>
        </div>
        <img className='Image' src={images[this.imageLink]} />
        <div className='Like'>
          <i id='Click-like' onClick={this.likeClick} className={(this.state.liked ? "fa fa-heart" : "fa fa-heart-o")}></i>
          <span className='Likes'>{this.state.likes}</span>
        </div>
        <div className='Description'>{this.props.data.description}</div>
      </div>
    );
  }
}

export default PhotoContainer;
