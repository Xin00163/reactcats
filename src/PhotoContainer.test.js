import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import PhotoContainer from './PhotoContainer';
import photoCardsData from './photoCardsData.json'

describe('PhotoContainer', () => {
  it('renders and matches our snapshot', () => {
    const component = renderer.create(
      <PhotoContainer data={photoCardsData[0]} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the initial like count', () => {
    const wrapper = mount(<PhotoContainer data={photoCardsData[0]} />);
    expect(wrapper.find('.Likes').text()).toEqual('53');
  });
})

describe('Liked state', () => {
  it('should have class "fa-heart-o" if liked state is false', () => {
      const wrapper = mount(<PhotoContainer data={photoCardsData[1]} />);
      expect(wrapper.state('liked')).toEqual(false);
      expect(wrapper.find('#Click-like').hasClass('fa-heart-o')).toEqual(true);

    });

  it('should have class "fa-heart" if liked state is true', () => {
      const wrapper = mount(<PhotoContainer data={photoCardsData[0]} />);
      expect(wrapper.state('liked')).toEqual(true);
      expect(wrapper.find('#Click-like').hasClass('fa-heart')).toEqual(true);
    });
});

describe('LikeClick', () => {
  it('clicking on unliked post, update the likes count and change the state to like', () => {
      const wrapper = mount(<PhotoContainer data={photoCardsData[1]} />);
      wrapper.find('#Click-like').simulate('click');
      expect(wrapper.state('liked')).toEqual(true);
      expect(wrapper.find('.Likes').text()).toEqual('674');
    });

  it('clicking on liked post, reduce the likes count and change the state', () => {
      const wrapper = mount(<PhotoContainer data={photoCardsData[0]} />);
      wrapper.find('#Click-like').simulate('click');
      expect(wrapper.state('liked')).toEqual(false);
      expect(wrapper.find('.Likes').text()).toEqual('52');
  });
});
