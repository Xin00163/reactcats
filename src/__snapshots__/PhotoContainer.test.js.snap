// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoContainer renders and matches our snapshot 1`] = `
<div
  className="PhotoContainer"
>
  <button
    id="Click-delete"
    onClick={undefined}
  >
    Delete post
  </button>
  <div
    className="User"
  >
    <img
      className="Avatar"
      src={
        Object {
          "process": [Function],
        }
      }
    />
    <h2
      className="Username"
    >
      heacan
    </h2>
  </div>
  <img
    className="Image"
    src={
      Object {
        "process": [Function],
      }
    }
  />
  <div
    className="Like"
  >
    <i
      className="fa fa-heart"
      id="Click-like"
      onClick={[Function]}
    />
    <span
      className="Likes"
    >
      53
    </span>
  </div>
  <div
    className="Description"
  >
    A lie is like a cat: You need to stop it before it gets out the door or itâ€™s really hard to catch.
  </div>
</div>
`;
